pkg/app/piped/cmd/piped/piped.go

```diff
@@ -789,3 +804,12 @@ func loginToOCIRegistry(ctx context.Context, execPath, address, username, passwo
 	}
 	return nil
 }
+
+func deepcopy(dst interface{}, src interface{}) error {
+	b, err := json.Marshal(src)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(b, dst)
+}
```

Comment:
How about making it less generic?
For example, adding a `Clone` function to the config package.

``` golang
func (c *Config) Clone() (*Config, error) {
}

Reply:
I implemented with this commit: [9fc4cc9](https://github.com/pipe-cd/pipecd/pull/3673/commits/9fc4cc90b102a4ccef9f9d33cae05e0a39a932eb)

Reply:
Got it! That is fine.